File: Domain/Entities/Comment.cs
using Domain.Entities.OwnedEntities;

namespace Domain.Entities
{
    public class Comment: EntityBase
    {
        public string? Content { get; set; }
        public DateTime CommentedOn { get; set; }
        public int OwnerId { get; set; }
        public int? ReplyTo { get; set; }
        public int PostId { get; set; }

        public Media? Media { get; set; }
        public User Owner { get; set; }
        public Post Post { get; set; }
        public ICollection<Comment> Replies { get; set; }

    }
}

File: Client/src/app/models/comments/comment.ts
import { User } from "../users/user"
import { Media } from "../media"

export interface Comment {
    id: number
    replyTo: number | null
    content: string
    commentedOn: string
    media: Media
    owner: User
    replies: Comment[]
    totalRepliesCount: number
    postId: number
}
